### PR TITLE
ref: Move files to the streaming_kafka_consumer package

### PR DESCRIPTION
**Pull request by lynnagara**
Moves most of the files that will eventually form the streaming
consumer library into the streaming_kafka_consumer package.
Note the test files, streaming consumer factory and a few
other pieces have not yet been moved over.

### CODE REVIEW

File: snuba/utils/streams/batching.py [hunk 1]
```diff
@@ -15,13 +15,13 @@
     TypeVar,
 )
 
-from snuba.utils.metrics import MetricsBackend
-from snuba.utils.streams.processing.strategies.abstract import (
+from streaming_kafka_consumer.processing.strategies.abstract import (
```
Comment (fpacifici) [inline snuba/utils/streams/batching.py:18]: Doesn't this file move ?

File: snuba/utils/streams/profiler.py [hunk 2]
```diff
@@ -4,12 +4,11 @@
 from pathlib import Path
```
Comment (fpacifici) [inline snuba/utils/streams/profiler.py:4]: Up to you but I would consider providing the streaming library with a profiler. So I would move this one there.

File: tests/utils/streams/backends/test_kafka.py [hunk 3]
```diff
@@ -9,20 +9,20 @@
 
 import pytest
```
Comment (fpacifici) [inline tests/utils/streams/backends/test_kafka.py:10]: I think we should move the tests as well in their own module so they go into the library.

File: snuba/utils/streams/batching.py (hunk 1 above)
Comment (lynnagara) [reply snuba/utils/streams/batching.py:18]: I was on the fence about this one. I can move it over too.

File: snuba/utils/streams/profiler.py (hunk 2 above)
Comment (lynnagara) [reply snuba/utils/streams/profiler.py:4]: Will do

File: tests/utils/streams/backends/test_kafka.py (hunk 3 above)
Comment (lynnagara) [reply tests/utils/streams/backends/test_kafka.py:10]: I'm going to handle the tests as a follow up.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**